rally-domain/src/main/groovy/yakworks/rally/orgs/repo/ContactRepo.groovy

```diff
@@ -93,6 +93,12 @@ class ContactRepo extends LongIdGormRepo<Contact> {
         if (data == null) data = [:] //if null then make it empty map so it can cycle down and blow error
 
         String sourceId = Maps.value(data, 'sourceId')
+        if(!data.sources && sourceId) data['sources'] = [[sourceId:sourceId]]
```

Comment:
this is not right, see the OrgRepo, that should be the model for how this is done.